src/traces/choropleth/hover.js

```diff
@@ -79,7 +79,14 @@ function makeHoverInfo(pointData, trace, pt, axis) {
     }
 
     if(hasZ) text.push(formatter(pt.z));
-    if(hasText) text.push(pt.tx);
+    if(hasText) {
+        var tx;
+
+        if(pt.tx) tx = pt.tx;
+        else if(trace.text) tx = trace.text;
+
+        if(!Array.isArray(tx)) text.push(tx);
```

Comment:
nice careful logic - `tx` could be an array if `trace.text` is too short to populate `pt.tx` or if it has falsey elements (hmm, what about `0`?)

This seems like a bit of logic that's repeated a bunch - and is probably done wrong in some places where it's *not* repeated... worth abstracting?

Reply:
Worth it for sure. Shouldn't be too bad.

Reply:
(the hardest part will be to figure out how to call this new function)

Reply:
See bc8294d and b637179 for what I came up with.

Reply:
Ooops I forget to update `choropleth/hover.js`, now done in -> 7f2604a6987925e0f211fe1f35b0821f0832dafb